src/nvim/terminal.c

```diff
@@ -307,26 +309,68 @@ static int on_osc(int command, VTermStringFragment frag, void *user)
     return 1;
   }
 
-  StringBuilder request = KV_INITIAL_VALUE;
-  kv_printf(request, "\x1b]%d;", command);
-  kv_concat_len(request, frag.str, frag.len);
-  schedule_termrequest(term, request.items, request.size);
+  if (frag.initial) {
+    if (kv_size(term->termrequest_buffer) > 0) {
+      kv_drop(term->termrequest_buffer, kv_size(term->termrequest_buffer));
```

Comment:
Same thing but imo a little nicer to read.

```suggestion
      kv_size(term->termrequest_buffer) = 0;
```